src/main/java/pl/uj/edu/JImageStream/api/ParallelBoundedImageTransform.java

```diff
@@ -1,45 +1,58 @@
 package pl.uj.edu.JImageStream.api;
 
 import java.awt.Point;
+import java.awt.image.BufferedImage;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Predicate;
 
 //todo fix this grebosz
 public class ParallelBoundedImageTransform implements ImageTransform {
 
+    //todo create default filter for whole image
+    //todo list of predicate
     private int height;
     private int width;
+    private BufferedImage image;
+    private Predicate<Point> predicate;
+    private Filter filter;
+    private ColorChannel[] colorChannels;
+    private int numberOfThreads;
 
-
-    public ParallelBoundedImageTransform(int height, int width, Predicate<Point> predicate, Filter filter) {
-        this.height = height;
-        this.width = width;
+    public ParallelBoundedImageTransform(BufferedImage bufferedImage, Predicate<Point> predicate, Filter filter, ColorChannel[] colorChannels, int numberOfThreads){
+        this.height = bufferedImage.getHeight();
+        this.width = bufferedImage.getWidth();
         this.predicate = predicate;
         this.filter = filter;
+        this.image = bufferedImage;
+        this.colorChannels = colorChannels;
+        this.numberOfThreads = numberOfThreads;
     }
 
-    //todo create default filter for whole image
-    //todo list of predicate
-    private Predicate<Point> predicate;
-    private Filter filter;
-
-
     @Override
     public void apply() {
         //todo set executors from stream
-        ExecutorService executor = Executors.newFixedThreadPool(1000);
+        ExecutorService executor = Executors.newFixedThreadPool(numberOfThreads);
```

Comment:
use min(numberOFThreads, width)? no reason to create more threads than we can use